viewing_party/party.py

```diff
@@ -0,0 +1,192 @@
+
+    for movie in range(len(user_watched)):
+        for x in range(len(friends_watched)): 
+            for item in friends_watched[x]["watched"]:
+                if user_watched[movie]["title"] in item.values():
+                    not_unique_titles.append(item)
+
+    for title in user_watched:
+        for movie in not_unique_titles:
+            if movie in user_watched:
+                user_watched.remove(movie)
+    return user_watched
+
+def get_friends_unique_watched(user_data):
+    """
+    Returns a list of dictionaries that represents a list of movies that are in the user's 
+    friends watched list(s), but are not in the user's watched list. Friend(s) have watched movie, user has not.
+    """
+    friends_watched_titles = []
+    user_watched = user_data["watched"] 
+    friends_watched = user_data["friends"]
+
+    # Adds every movie in each friend's watched list to friends_watched_titles list
+    for x in range(len(friends_watched)): 
+        for item in friends_watched[x]["watched"]:
+            if item not in friends_watched_titles:  # to not duplicate a movie already added in a previous iteration
+                friends_watched_titles.append(item)
+
+    # Removes a movie found in the user's watched list from the friends_watched_titles list
+    for title in user_watched:
+        for movie in friends_watched_titles:
+            if title["title"] in movie.values():
+                friends_watched_titles.remove(movie)
+    return friends_watched_titles
+
+# ——— WAVE 04 ———
+def get_available_recs(user_data):
+    """
+    Returns a list of dictionaries that represents a user's recommended movies. Calls the get_friends_unique_watched function
+    to get movies that the user has not watched, but at least one of the user's friends has watched. Checks if each movie is 
+    available in a subscription service that the user has, and only includes these movies in the returned recommendations list.
+    """
+    recommendations = get_friends_unique_watched(user_data)
+
+    for i in range(len(recommendations)):
+        for recommendation in recommendations:
+            if recommendation["host"] not in user_data["subscriptions"]:
+                recommendations.remove(recommendation)
+    return recommendations
+
+# ——— WAVE 05 ———
+def get_new_rec_by_genre(user_data):
+    """
+    Returns a list of dictionaries that represents a user's recommended movies based on the user's most watched genre, 
+    and movies of that genre that user's friend(s) have watched but the user has not. 
+    """
+    fave_genre = get_most_watched_genre(user_data)
+    recommendations = get_friends_unique_watched(user_data)
+
+    if len(user_data["watched"]) == 0:
+        return user_data["watched"]
+
+    for recommendation in recommendations:
+        if recommendation["genre"] != fave_genre:
+            recommendations.remove(recommendation)
+    return recommendations 
+
+def get_rec_from_favorites(user_data):
```

Comment:
👍 this works, but it seems like it's a bit long.  Perhaps it could be simplified.